tests/build.sh

```diff
@@ -4,115 +4,65 @@ set -exuo pipefail
-OS_ID=$(echo "$TEST_OS" | cut -d '-' -f 1)
-OS_VERSION_ID=$(echo "$TEST_OS" | cut -d '-' -f 2)
-
-# Base image
-case "$OS_ID" in
-    "centos")
-        TIER1_IMAGE_URL="quay.io/centos-bootc/centos-bootc:stream${OS_VERSION_ID}"
+# If provided should be of the form fedora-42 or centos-10
+target=${1:-}
+
+build_args=()
+if test -n "${target:-}"; then
+    shift
+    # Get OS info from TEST_OS env
+    OS_ID=$(echo "$target" | cut -d '-' -f 1)
+    OS_VERSION_ID=$(echo "$target" | cut -d '-' -f 2)
+
+    # Base image
+    case "$OS_ID" in
+        "centos")
+            BASE="quay.io/centos-bootc/centos-bootc:stream${OS_VERSION_ID}"
         ;;
-    "fedora")
-        TIER1_IMAGE_URL="quay.io/fedora/fedora-bootc:${OS_VERSION_ID}"
+        "fedora")
+            BASE="quay.io/fedora/fedora-bootc:${OS_VERSION_ID}"
         ;;
-esac
-
-CONTAINERFILE="${BOOTC_TEMPDIR}/Containerfile"
-tee "$CONTAINERFILE" > /dev/null << CONTAINERFILEOF
-FROM $TIER1_IMAGE_URL as build
-
-WORKDIR /code
-
-RUN <<EORUN
-set -xeuo pipefail
-. /usr/lib/os-release
-case \$ID in
-    centos|rhel) dnf config-manager --set-enabled crb;;
-    fedora) dnf -y install dnf-utils 'dnf5-command(builddep)';;
-esac
-dnf -y builddep contrib/packaging/bootc.spec
-dnf -y install git-core
-EORUN
-
-RUN mkdir -p /build/target/dev-rootfs
-# git config --global --add safe.directory /code to fix "fatal: detected dubious ownership in repository at '/code'" error
-RUN --mount=type=cache,target=/build/target --mount=type=cache,target=/var/roothome git config --global --add safe.directory /code && make test-bin-archive && mkdir -p /out && cp target/bootc.tar.zst /out
-
-FROM $TIER1_IMAGE_URL
-
-# Inject our built code
-COPY --from=build /out/bootc.tar.zst /tmp
-RUN tar -C / --zstd -xvf /tmp/bootc.tar.zst && rm -vrf /tmp/*
-
-RUN <<EORUN
-set -xeuo pipefail
-
-# Provision test requirement
-/code/hack/provision-derived.sh
-# Also copy in some default install configs we use for testing
-cp -a /code/hack/install-test-configs/* /usr/lib/bootc/install/
-# And some test kargs
-cp -a /code/hack/test-kargs/* /usr/lib/bootc/kargs.d/
-
-# For testing farm
-mkdir -p -m 0700 /var/roothome
-
-# Enable ttyS0 console
-mkdir -p /usr/lib/bootc/kargs.d/
-cat <<KARGEOF >> /usr/lib/bootc/kargs.d/20-console.toml
-kargs = ["console=ttyS0,115200n8"]
-KARGEOF
-
-# For test-22-logically-bound-install
-cp -a /code/tmt/tests/lbi/usr/. /usr
-ln -s /usr/share/containers/systemd/curl.container /usr/lib/bootc/bound-images.d/curl.container
-ln -s /usr/share/containers/systemd/curl-base.image /usr/lib/bootc/bound-images.d/curl-base.image
-ln -s /usr/share/containers/systemd/podman.image /usr/lib/bootc/bound-images.d/podman.image
-
-# Install rsync which is required by tmt
-dnf -y install cloud-init rsync
-dnf -y clean all
+        *) echo "Unknown OS: ${OS_ID}" 1>&2; exit 1
+        ;;
+    esac
+    build_args+=("--build-arg=base=$BASE")
+fi
 
-rm -rf /var/cache /var/lib/dnf
-EORUN
-CONTAINERFILEOF
+just build ${build_args[@]}
```

Comment:
For local running, the `just` has to be installed locally, right?

Reply:
Yeah I'm trying to centralize more of the dev/test entrypoint around `just`